baseapp/baseapp.go

```diff
@@ -563,11 +564,17 @@ func (app *BaseApp) runTx(mode runTxMode, txBytes []byte, tx sdk.Tx) (gInfo sdk.
 			ctx = newCtx.WithMultiStore(ms)
 		}
 
+		events = ctx.EventManager().Events()
+
 		// GasMeter expected to be set in AnteHandler
 		gasWanted = ctx.GasMeter().Limit()
 
 		if err != nil {
-			return gInfo, nil, err
```

Comment:
This isn't necessary. Upon error, Check/DeliverTx will only use the gas info and the error. Please correct me if I'm wrong.

Reply:
you are right, should we add the events in case of an error? cc: @zmanian @jackzampolin

Reply:
I don't think we need to emit events for txs that don't succceed/